pwa-shell/src/ui/content-list/ContentList.js

```diff
@@ -27,19 +27,33 @@ class ContentList extends PureComponent {
   }
 
   componentDidMount() {
-    this.props.fetchContentList();
+    const { fetchContentList, isSearchResult, searchTerms } = this.props;
+    const filters = { searchTerms: isSearchResult ? searchTerms : '' };
+    fetchContentList(filters);
   }
 
   componentDidUpdate(prevProps) {
     if (this.props.contentType !== prevProps.contentType) {
-      this.props.fetchContentList();
+      const { fetchContentList, isSearchResult, searchTerms } = this.props;
```

Comment:
what about extracting the duplicated code between `componentDidMount` and `componentDidUpdate`?